x-pack/plugins/siem/common/endpoint/types.ts

```diff
@@ -6,11 +6,20 @@
 
 import { Datasource, NewDatasource } from '../../../ingest_manager/common';
 
+/**
+ * Object that allows you to maintain stateful information in the location object across navigation events
+ *
+ */
+
 export interface AppLocation {
   pathname: string;
   search: string;
   hash: string;
   key?: string;
+  // Other parts of the application use the 'history' package which defines the state as type any.
+  // Also see: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/27012
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  state?: any;
```

Comment:
why not 
```ts
// Used by whatever this PR does
state?: { isTabChange?: boolean }
```

Reply:
Any defined shapes for state are going to conflict with the types defined by  the `history` package being used here.

https://github.com/elastic/kibana/blob/8ad9f442b7b43879715260b5b7b790279b0f8a9a/x-pack/plugins/siem/public/endpoint_alerts/store/alert_list_pagination.test.ts#L52

Same issue referenced here: 
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/27012

I thought about either replacing our `AppLocation` with the `LocationDescriptorObject` or at the very least aliasing it, but wanted your opinion on that as `LocationDescriptorObject` defines every parameter in the location object as optional. The only place where I see it used in Kibana is here:

https://github.com/elastic/kibana/blob/8ad9f442b7b43879715260b5b7b790279b0f8a9a/x-pack/plugins/siem/public/common/hooks/endpoint/use_navigate_by_router_event_handler.ts#L21

Reply:
Thanks for the reply. How about this: https://github.com/oatkiller/kibana/commit/44cb906e3de6c02a8e07b02c46f97802a32f5d29

I set the type for `history` in those tests like:
```
let history: History<never>;
```
because at the time we didn't use state (and I hadn't ever used it or seen it used) and because I didn't know TS that well at the time.